File: frontend/src/types/message.ts
export type Message = {
  id: string;
  senderId: string;
  message: string;
  isEdited: boolean;
  createdAt: string;
  senderUsername: string;
  senderImage: string;
  channelId?: string;
  recipientId?: string;
  replyToId?: string;
  replyToUsername?: string | null;
  replyToImage?: string | null;
  replyToMessage?: string | null;
};

File: backend/models/message.go
package models

import "time"

type DirectMessage struct {
	ID                string    `json:"id"`
	SenderID          string    `json:"senderId"`
	RecipientID       string    `json:"recipientId"`
	Message           string    `json:"message"`
	HiddenBySender    bool      `json:"hiddenBySender"`
	HiddenByRecipient bool      `json:"hiddenByRecipient"`
	IsEdited          bool      `json:"isEdited"`
	CreatedAt         time.Time `json:"createdAt"`
	ReplyToID         *string   `json:"replyToId"`
	SenderUsername    string    `json:"senderUsername"`
	SenderImage       *string   `json:"senderImage"`
	ReplyToUsername   *string   `json:"replyToUsername"`
	ReplyToImage      *string   `json:"replyToImage"`
	ReplyToMessage    *string   `json:"replyToMessage"`
}
